### PR TITLE
added new PGE called AOI merged track stitcher

### PR DESCRIPTION
**Pull request by DustinKLo**
added hysds.io and job-spec json files
removed all other arguments in job-spec except "extra_products" in job-spec and python script
included "dataset_jpath:_source.metadata.s1-gunw_urls" in extra_products.from

### CODE REVIEW

File: interferogram/aoi_merged_track_stitcher.py [hunk 1]
```diff
@@ -0,0 +1,355 @@
...
+    print('generated met.json object: {}'.format(json.dumps(met, indent=2)))
+
+    with open(met_json_filename, 'w') as f:  # writing the met.json
+        met_json = json.dumps(met, indent=2)
+        f.write(met_json)
+    return True
+
+
+if __name__ == '__main__':
+    VERSION = 'v2.0'  # values needed for the dataset.json file
```
Comment (pymonger) [inline interferogram/aoi_merged_track_stitcher.py]: The versions of datasets created in this repo is controlled by the `dataset_versions.json` under conf: https://github.com/aria-jpl/ariamh/blob/develop/conf/dataset_versions.json. You can add it there and pull from it to get the version. Example of doing that here: https://github.com/aria-jpl/ariamh/blob/28b0ac9d7fb7242717e7dfe7f247457767e95812/interferogram/stitch_ifgs.py#L24-L33.

**Review by pymonger — CHANGES_REQUESTED** [review]

**Comment by DustinKLo** [comment]
all files created when PGE completes
```
-rw-r--r-- 1 ops ops       227 May 31 18:31 aoi_merged_track_stitcher.log
-rw-r--r-- 1 ops ops  33613839 May 31 18:40 filt_topophase.unw.conncomp.geo
-rw-r--r-- 1 ops ops       549 May 31 18:40 filt_topophase.unw.conncomp.geo.vrt
-rw-r--r-- 1 ops ops      4388 May 31 18:40 filt_topophase.unw.conncomp.geo.xml
-rw-r--r-- 1 ops ops 268910712 May 31 18:40 filt_topophase.unw.geo
-rw-r--r-- 1 ops ops   1425924 May 31 18:41 filt_topophase.unw.geo.browse.png
-rw-r--r-- 1 ops ops    202311 May 31 18:41 filt_topophase.unw.geo.browse_small.png
-rw-r--r-- 1 ops ops       875 May 31 18:40 filt_topophase.unw.geo.vrt
-rw-r--r-- 1 ops ops      4381 May 31 18:40 filt_topophase.unw.geo.xml
-rw-r--r-- 1 ops ops       647 May 31 18:40 inputs.json
-rw-r--r-- 1 ops ops         0 May 31 18:31 isce.log
-rw-r--r-- 1 ops ops       853 May 31 18:40 los.rdr.geo.vrt
-rw-r--r-- 1 ops ops      4625 May 31 18:40 los.rdr.geo.xml
-rw-r--r-- 1 ops ops 134455356 May 31 18:40 phsig.cor.geo
-rw-r--r-- 1 ops ops       535 May 31 18:40 phsig.cor.geo.vrt
-rw-r--r-- 1 ops ops      4211 May 31 18:40 phsig.cor.geo.xml
-rw-r--r-- 1 ops ops      1940 May 31 18:41 S1-GUNW-MERGED_TN014_20190224T152533-20190320T152714-poeorb-6542.dataset.json
-rw-r--r-- 1 ops ops      1734 May 31 18:41 S1-GUNW-MERGED_TN014_20190224T152533-20190320T152714-poeorb-6542.met.json
```

**Comment by DustinKLo** [comment]
filt_topophase unw geo browse_small

**Comment by DustinKLo** [comment]
@pymonger  looks like this regex works
`S1-GUNW-MERGED_TN.*?_.*?-(?P<year>\d{4})(?P<month>\d{2})(?P<day>\d{2})(?P<rest>.+)`

File: interferogram/aoi_merged_track_stitcher.py (hunk 1 above)
Comment (DustinKLo) [reply interferogram/aoi_merged_track_stitcher.py]: pushed changes

**Comment by DustinKLo** [comment]
confluence wiki for the whole PGE: https://aria.atlassian.net/wiki/spaces/ARIA/pages/2424838/AOI+Merged+Track+Stitcher

**Review by pymonger — APPROVED** [review]

File: interferogram/aoi_merged_track_stitcher.py [hunk 2]
```diff
@@ -0,0 +1,364 @@
...
+
+def get_version():
+    # dataset version: stored in conf/dataset_versions.json
+    ds_vers_cfg = os.path.normpath(
+        os.path.join(
+            os.path.dirname(os.path.abspath(__file__)),
+            '..', 'conf', 'dataset_versions.json'))
+    with open(ds_vers_cfg) as f:
+        ds_vers = json.load(f)
+    return ds_vers['S1-IFG-STITCHED']
```
Comment (pymonger) [inline interferogram/aoi_merged_track_stitcher.py]: @DustinKLo the dataset shortname should be https://github.com/aria-jpl/ariamh/blob/standard-product/conf/dataset_versions.json#L8 however that only exists in the `standard-product` branch. I think we need @mkarim2017 to create a pull request first to merge in `standard-product` into `develop` then we can process this pull request. For now, fix this here and we'll need to sort out the `standard-product` vs `develop` branch when @mkarim2017 gets back.

**Review by pymonger — CHANGES_REQUESTED** [review]

**Review by pymonger — APPROVED** [review]

**Pull request merged**